src/solvers/refinement/string_constraint_generator_valueof.cpp

```diff
@@ -234,35 +238,34 @@ string_exprt string_constraint_generatort::add_axioms_from_int_with_radix(
     // a6 : |res| == size && res[0] is a digit for radix => input_int == sum
     // a7 : |res| == size && res[0] == '-' => input_int == -sum
 
-    exprt::operandst digit_constraints;
-    exprt sum=
-      get_numeric_value_from_character(res[0], char_type, type, radix_ul);
-
-    for(size_t j=1; j<size; j++)
+    if(size>1)
```

Comment:
The documentation above (`sum_j`, `a5`) must be updated if we no longer iterate over `j`. At the moment I'm having difficulties evaluating the soundness of this new method. Do you mind updating the doc before I come back to reviewing this?